### PR TITLE
Master lps 193721 | Add functional tests for generate path structure for api endpoints

### PR DESCRIPTION
**Pull request by yanan**
**Background**:
- Add functional tests on generate path structure for endpoints

**Note**:
- `CanCopyUrlOfEndpoint` is already automatated in ListAPIApplicationEndpoints.CanCopyURLOfAPIEndpoint, not going to automate it again in this pr
- Integration test for `CanValidatePathThroughAPI` is already added in APIEndpointRelevantObjectEntryModelListenerTest.test, not going to add it in this pr
- Update to use api to create api application in Given + And Given

**Jira Ticket**:
- https://liferay.atlassian.net/browse/LPS-192114

**Test Plan**:
- Given going to Control Panel > Object > Headless BuilderWhen I create a new API ApplicationThen I can see label under url is {protocol}//{host}:{port}/o/c
- Given creating an API application in Control Panel > Object > Headless BuilderWhen Edit the API applicationThen I can see label under url is {protocol}//{host}:{port}/o/c
- Given creating an API application in Control Panel > Object > Headless BuilderAnd Given publish the API application by Change Publication StatusWhen I can to API ExplorerThen the c/{url} is present in Rest Applications
- Given creating an API application in Control Panel > Object > Headless BuilderAnd Given publishing the API applicationWhen I navigate to {protocol}//{host}:{port}/o/api?endpoint={protocol}//{host}:{port}/o/c/{applicationUrl}/openapi.jsonThen the page can load without errors
- Given creating an API application in Control Panel > Object > Headless BuilderAnd Given publishing the API application by Change Publication StatusWhen I navigate to /o/c/{applicationUrl}Then {protocol}//{host}:{port}/o/c/{applicationUrl}/openapi.json is presentAnd Then {protocol}//{host}:{port}/o/c/{applicationUrl} is present in Servers
- Given with postAPIApplication including apiApplicationToAPIEndpoints and path to create an API application with its endpointAnd Given going to Control Panel > Object > Headless BuilderWhen I navigate to Endpoints tab in Edit API applicationThen I can see correct path
- Given creating an API application in Control Panel > Object > Headless BuilderAnd Given setting Panel Category Key to Control Panel > Object for API EndpointWhen I create an endpoint with id of API application and path not start with '/'Then I can see error messageAnd Then the endpoint is not being created
- Given creating an API application in Control Panel > Object > Headless BuilderAnd Given setting Panel Category Key to Control Panel > Object for API EndpointWhen I create an endpoint with id of API application and valid pathThen I can see endpoint is created with correct path
- Given with postAPIApplication including apiApplicationToAPIEndpoints and path to create an published API application with its endpointWhen I navigate to /o/c/{applicationUrl}Then I can see '/{path}' is present under get{Endpoint}Page

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:sf

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path
Comment (Hijky) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path:1]: Just FYI, I think we need to add the new path to sort in ascending order, and I have done some changes in this [PR](https://github.com/liferay-headless/liferay-portal/pull/1444/files)

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GeneratePathStructureForEndpoints.testcase [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@priority = 4
+	test APIUrlIsPresentInAPIExplorerRestApplications {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
```
Comment (Hijky) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GeneratePathStructureForEndpoints.testcase:32]: I'm don't make sure if we can move 'property test.run.type = "single";' to global setting

**Review by Hijky — COMMENTED** [review]
The others are good!

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path
Comment (yanan) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path:1]: That's good, thanks for the note.

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GeneratePathStructureForEndpoints.testcase (hunk 1 above)
Comment (yanan) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GeneratePathStructureForEndpoints.testcase:32]: I think it's better to keep it same with current master, so I'm going to update this, thanks.

**Comment by yanan** [comment]
Failures are not related, and the failed test have been improved in another ticket
image

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GeneratePathStructureForEndpoints.testcase [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+		task ("Then the c/{url} is present in Rest Applications") {
+			AssertElementPresent(
+				locator1 = "Button#BUTTON_WITH_VALUE",
+				value = "c/app-test");
+		}
+	}
+
+	@priority = 3
+	test APIUrlsArePresentInAPIExplorer {
```
Comment (magjed4289) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GeneratePathStructureForEndpoints.testcase:51]: just a reminder that test with priority fp3 are not required in the first phase of automation :)

**Review by magjed4289 — APPROVED** [review]

**Comment by yanan** [comment]
ci:forward

**Pull request closed without merging**